contracts/examples/OracleCreator.sol

```diff
@@ -0,0 +1,183 @@
+    mapping(uint256 => Oracle) public oracles;
+    uint256 public oraclesIndex;
+    mapping (uint256 => Observation[]) public oracleObservations;
+    
+    constructor(
+        address payable _DXswapRelayer
+    ) public {
+        DXswapRelayer = _DXswapRelayer;
+    }
+
+    function createOracle(
+        uint256 windowSize,
+        address pair,
+        uint8 granularity,
+        uint256 executionBountyWei
+    ) public payable returns (uint256 oracleId) {
+        uint256 periodSize;
+        require(msg.sender == DXswapRelayer, 'DXliquidityRelay: CALLER_NOT_RELAYER');
+        require(granularity > 1, 'DXliquidityRelay: GRANULARITY');
+        require(
+            (periodSize = windowSize / granularity) * granularity == windowSize,
+            'DXliquidityRelay: WINDOW_NOT_EVENLY_DIVISIBLE'
+        );
+        require(msg.value >= executionBountyWei.mul(periodSize), 'DXliquidityRelay: INVALID_BOUNTY');
+
+        oracles[oraclesIndex] = Oracle(
+            windowSize,
+            pair,
+            granularity,
+            periodSize,
+            executionBountyWei,
+            0,
+            false
+        );
+
+      oracleId = oraclesIndex;
+      emit OracleCreated(oracleId, pair, windowSize, granularity, periodSize, executionBountyWei);
+      oraclesIndex++;
+    }
+
+    function update(uint256 _oracleId) public payable {
+        require(_oracleId <= oraclesIndex, 'DXliquidityRelay: INVALID_ORACLEID');
+        require(
+            oracles[_oracleId].observationFinished == false,
+            'DXliquidityRelay: OBSERVATION_FINISHED'
+        );
+
+        address pair = oracles[_oracleId].pair;
+
+        // populate the array with empty observations (first call only)
+        for (uint256 i = oracleObservations[_oracleId].length; i < oracles[_oracleId].granularity; i++) {
+            oracleObservations[_oracleId].push();
+        }
+
+        // get the observation for the current period
+        uint8 observationIndex = observationIndexOf(_oracleId, block.timestamp);
+        Observation storage observation = oracleObservations[_oracleId][observationIndex];
+
+        // we only want to commit updates once per period (i.e. windowSize / granularity)
+        uint256 timeElapsed = block.timestamp - observation.timestamp;
+        if (timeElapsed > oracles[_oracleId].periodSize) {
+            (uint256 price0Cumulative, uint256 price1Cumulative, ) = DXswapOracleLibrary.currentCumulativePrices(pair);
+            observation.timestamp = block.timestamp;
+            observation.price0Cumulative = price0Cumulative;
+            observation.price1Cumulative = price1Cumulative;
+        }
+
+        if (observationIndex == oracleObservations[_oracleId].length) {
+            oracles[_oracleId].observationFinished = true;
+            oracles[_oracleId].finishedTimestamp = block.timestamp;
+            emit OracleExecuted(_oracleId);
+        }
+
+        // send the bounty to msg.sender
```

Comment:
We should add the check to only try to send when balance of the oracle is enough to send it. Allowing also reward of 0 ETH, this might happen and later a reward can be requested to DXdao or sent form dxdao to addresses that executed the updates.

Also, are you shoule keep track of the remaining bounty balance of each oracle separated, right? you are just storing the bounty per update, we can assume that an oracle ran out of balance but we still want to update it and leave the option for it.

And about external rewards to "oracle updaters" this can easily be done by filtering the update calls execuetd succesfully, just throwing this in to be consider for the future.